hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java

```diff
@@ -168,15 +168,15 @@ protected List<TokenTransferList> asTokenTransferListFrom(@NonNull final Map<Ent
      */
     @NonNull
     protected Map<TokenID, List<NftTransfer>> nftChangesFrom(
-            @NonNull final WritableNftStore writableNftStore, @NonNull final ReadableTokenStore readableTokenStore) {
+            @NonNull final WritableNftStore writableNftStore, @NonNull final WritableTokenStore writableTokenStore) {
         final var nftChanges = new HashMap<TokenID, List<NftTransfer>>();
         for (final NftID nftId : writableNftStore.modifiedNfts()) {
             final var modifiedNft = writableNftStore.get(nftId);
             final var persistedNft = writableNftStore.getOriginalValue(nftId);
 
             // The NFT may not have existed before, in which case we'll use a null sender account ID
             AccountID senderAccountId = null;
-            final var token = requireNonNull(readableTokenStore.get(nftId.tokenIdOrThrow()));
+            final var token = requireNonNull(writableTokenStore.get(nftId.tokenIdOrThrow()));
```

Comment:
Wondering why you change to writableTokenStore here as you only use the token value and not update it, readableTokenStore didn't have the same token value?

Reply:
@iwsimon yes `writableTokenStore` is added to go through modified tokens and see if treasury changed in `nftChangesFrom`